framework/src/controller/channels/child_process/index.js

```diff
@@ -0,0 +1,10 @@
+const setupProcessHandlers = channel => {
+	process.once('SIGTERM', () => channel.cleanup(1));
+	process.once('SIGINT', () => channel.cleanup(1));
+	process.once('cleanup', (error, code) => channel.cleanup(code, error));
+	process.once('exit', (error, code) => channel.cleanup(code, error));
+};
+
+module.exports = {
+	setupProcessHandlers,
+};
```

Comment:
this file should have a test too.

Reply:
Added skeleton for these tests and will be addressed in another issue.